src/components/MetricChart/MetricChart.js

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { Chart } from 'chart.js/auto'
+import classnames from 'classnames'
+
+import Loader from '../../common/Loader/Loader'
+
+import { CHART_TYPE_BAR } from '../../constants'
+
+import { calculateMaxTicksLimit, generateMetricChartTooltip } from './metricChart.util'
+
+const GenericMetricChart = ({ chartConfig, showGrid }) => {
+  const chartRef = useRef(null)
+  const chartInstance = useRef(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const canvasClassNames = classnames(isLoading && 'hidden-canvas')
+  const customPoints = useMemo(() => {
+    const hasDriftStatusList = chartConfig.data.datasets[0]?.driftStatusList?.length !== 0
+    const totalDriftIndex = chartConfig.data.datasets[0]?.totalDriftStatus?.index
+
+    return {
+      radius: context => {
+        const isCurrentIndexTotalDriftIndex = context.dataIndex === totalDriftIndex
+        return hasDriftStatusList && isCurrentIndexTotalDriftIndex ? 2 : 0
+      },
+      pointStyle: context => (hasDriftStatusList ? 'circle' : 'none'),
+      backgroundColor: () => (hasDriftStatusList ? 'black' : undefined),
```

Comment:
if possible replace undefined with null

Reply:
with null getting error:
Uncaught TypeError: Cannot read properties of null (reading 'a')